### PR TITLE
Add jenkins extension

### PR DESCRIPTION
**Pull request by k8scat**
## Description

Add jenkins extension

## Screencast

image

image

image

image

image

image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by k8scat** [comment]
@pernielsentikaer this extension can be reviewed, thanks

File: extensions/jenkins/src/jenkins.tsx [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+  useEffect(() => {
+    search("");
+  }, []);
+
+  return (
+    <List
+      isLoading={isLoading}
+      onSearchTextChange={search}
+      searchBarPlaceholder="Search jenkins..."
```
Comment (pernielsentikaer) [inline extensions/jenkins/src/jenkins.tsx:46]:
```suggestion
      searchBarPlaceholder="Search Jenkins..."
```

File: extensions/jenkins/src/jenkins.tsx [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+      throttle
+      actions={
+        <ActionPanel>
+          <Action.Push
+            icon={Icon.NewDocument}
+            title="Add jenkins"
+            target={<AddJenkins setJenkinsList={setJenkinsList} />}
+          />
+        </ActionPanel>
```
Comment (pernielsentikaer) [inline extensions/jenkins/src/jenkins.tsx:55]: The confusing part 🙂

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @k8scat 👋 

Thanks for this extension; I gave it a spin and had some feedback:

Instead of having your preferences, it would make much more sense to use [preferences](https://developers.raycast.com/basics/prepare-an-extension-for-store#preferences) for this - could you look into this instead? It's pretty confusing as it is right now 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/jenkins/src/jenkins.tsx (hunk 2 above)
Comment (k8scat) [reply extensions/jenkins/src/jenkins.tsx]: In my mind, Preferences can only be used to set one account information, but this extension is being to manage multi jenkins acounts

Comment (k8scat) [reply extensions/jenkins/src/jenkins.tsx]: Because it could not add jenkins with empty list, this is why I add actions at List

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**